qiita_ware/ebi.py

```diff
@@ -182,14 +195,25 @@ def __init__(self, preprocessed_data_id, action):
                 platform = sample_prep['platform'].upper()
                 if platform not in self.valid_platforms:
                     nvp.append(k)
+                else:
+                    if 'instrument_model' not in sample_prep:
+                        nvim.append(k)
+                    else:
+                        im = sample_prep['instrument_model'].upper()
+                        if im not in self.valid_platforms[platform]:
+                            nvim.append(k)
 
             self.samples[k] = v
             self.samples_prep[k] = sample_prep
             self.sample_demux_fps[k] = get_output_fp("%s.fastq.gz" % k)
 
         if nvp:
-            error_msgs.append("These samples do not have a valid platform: "
-                              "%s" % (', '.join(nvp)))
+            error_msgs.append("These samples do not have a valid platform "
+                              "(instrumet model wasn't checked): %s" % (
+                                ', '.join(nvp)))
+        if nvim:
```

Comment:
Can you add a test case in `test_ebi.py` for the cases where there are invalid instrument models?